### PR TITLE
Update scriptcanvas nodeable node doc

### PR DESCRIPTION
**Pull request by onecent1101**
## Change summary
* Update scriptcanvas nodeable doc with new registry guide
* Update scriptcanvas nodeable doc with easy example, and move Timer node example to advanced topic

### Submission Checklist:

* [ ] **Descriptive active voice** - Do descriptive sentences have a clear *subject* and *action verb*?
* [ ] **Answer the question at hand** - Does the documentation answer a *what*, *why*, *how*, or *where* type of question?
* [ ] **Consistency** - Does the content consistently follow the [style guide](https://o3de.org/docs/contributing/to-docs/style-guide/)?
* [ ] **Help the user** - Does the documentation show the user something *meaningful*?

Signed-off-by: onecent1101 <liug@amazon.com>

### CODE REVIEW

File: content/docs/user-guide/scripting/script-canvas/programmer-guide/custom-nodes/custom-nodeable-nodes.md [hunk 1]
```diff
@@ -168,88 +140,163 @@ There are three critical parts that every Script Canvas nodeable header file nee
...
+        m_descriptors.insert(m_descriptors.end(), autogenDescriptors.begin(), autogenDescriptors.end());
     }
     ```
+
+## Advanced ScriptCanvasNodeable.xml Usage
+This topic explores additional features we support in nodeable XML file.
+
+### Base and derived nodeable node
+If you have basic shared logic across multiple nodeable node, we do support create base and derived nodeable node.
+We'll use the following XML, copied from the O3DE source for the **Time Delay** node.
```
Comment (willihay) [inline content/docs/user-guide/scripting/script-canvas/programmer-guide/custom-nodes/custom-nodeable-nodes.md:246]:
```suggestion
If you have shared logic across multiple nodeable nodes, you can create a base node and multiple derived nodes.

The following example uses the O3DE source for the **Time Delay** node:
```

File: content/docs/user-guide/scripting/script-canvas/programmer-guide/custom-nodes/custom-nodeable-nodes.md [hunk 2]
```diff
@@ -168,88 +140,163 @@ There are three critical parts that every Script Canvas nodeable header file nee
...
+        </Input>
+
+        <Output Name="Done" Description="Signaled after waiting for the specified amount of times."/>
+
+        <PropertyInterface Property="m_timeUnitsInterface" Name="Units" Type="Input" Description="Units to represent the time in."/>
+    </Class>
+</ScriptCanvas>
+```
+
+The `TimeDelayNodeable` class itself implements a base class, called `BaseTimer`. In the following XML, you can see the base class defines shared properties
```
Comment (willihay) [inline content/docs/user-guide/scripting/script-canvas/programmer-guide/custom-nodes/custom-nodeable-nodes.md:270]:
```suggestion
The `TimeDelayNodeable` class implements a base class, called `BaseTimer`. In the following base class XML, you can see that the base class defines the shared properties, "Units" and "TickOrder":

File: [BaseTimer.ScriptCanvasNodeable.xml](https://github.com/o3de/o3de/blob/development/Gems/ScriptCanvas/Code/Include/ScriptCanvas/Internal/Nodeables/BaseTimer.ScriptCanvasNodeable.xml)

```

File: content/docs/user-guide/scripting/script-canvas/programmer-guide/custom-nodes/custom-nodeable-nodes.md [hunk 3]
```diff
@@ -168,88 +140,163 @@ There are three critical parts that every Script Canvas nodeable header file nee
...
+
+        <Property Name="m_tickOrder" Type="int" DefaultValue="static_cast&lt;int&gt;(AZ::TICK_DEFAULT)" Serialize="true">
+            <PropertyData Name="TickOrder" Description="When the tick for this time update should be handled."/>
+        </Property>
+    </Class>
+</ScriptCanvas>
+```
+
+{{< note >}}
+For further node name, tooltip and category customization, please refer to [Text Replacement](/docs/user-guide/scripting/script-canvas/editor-reference/text-replacement/)
```
Comment (willihay) [inline content/docs/user-guide/scripting/script-canvas/programmer-guide/custom-nodes/custom-nodeable-nodes.md:301]:
```suggestion
For further node name, tooltip, and category customization, please refer to [Text Replacement](/docs/user-guide/scripting/script-canvas/editor-reference/text-replacement/).
```

**Review by willihay — CHANGES_REQUESTED** [review]

File: content/docs/user-guide/scripting/script-canvas/programmer-guide/custom-nodes/custom-nodeable-nodes.md [hunk 4]
```diff
@@ -28,6 +28,8 @@ As an example, here is partial definition of StartingPointInput Gem's `Code/CMak
    AUTOGEN_RULES
        *.ScriptCanvasNodeable.xml,ScriptCanvasNodeable_Header.jinja,$path/$fileprefix.generated.h
        *.ScriptCanvasNodeable.xml,ScriptCanvasNodeable_Source.jinja,$path/$fileprefix.generated.cpp
+       *.ScriptCanvasNodeable.xml,ScriptCanvasNodeableRegistry_Header.jinja,AutoGenNodeableRegistry.generated.h
```
[the remark this answers is not in the document]
Comment (onecent1101) [reply content/docs/user-guide/scripting/script-canvas/programmer-guide/custom-nodes/custom-nodeable-nodes.md:31]: Actually we don't expect customer to use ScriptCanvasGrammar, right now ScriptCanvasGrammar is all for our internal usage and it is not a nodeable node.

Plus StartingPointInput Gem ScriptCanvasGrammar usage is even for deperecation. (We don't have any other Gem as a good reference)

File: content/docs/user-guide/scripting/script-canvas/programmer-guide/custom-nodes/custom-nodeable-nodes.md [hunk 5]
```diff
@@ -168,88 +140,163 @@ There are three critical parts that every Script Canvas nodeable header file nee
...
-For an example on how to create your own library, use the **StartingPointInput** Gem from the O3DE source as a reference:
-
-`Gems/StartingPointInput/Code/Source/InputLibrary.h`
-`Gems/StartingPointInput/Code/Source/InputLibrary.cpp`
-
-Libraries help with node organization, but their primary function is to ensure all nodes are pulled in and reflected at the appropriate time.
-
-In the **Timer** nodeable example, the node is added to the Script Canvas Gem's **Time** library in: `Gems/ScriptCanvas/Code/Include/ScriptCanvas/Libraries/Time/Time.cpp`.
+The final step is to register and reflect the new node. To do this, you need to modify Gem [Module](/docs/user-guide/programming/gems/overview/) and [System Component](/docs/user-guide/programming/components/system-components/). Use the **StartingPointInput** Gem from the O3DE source as a reference:
+
```
[the remark this answers is not in the document]
Comment (onecent1101) [reply content/docs/user-guide/scripting/script-canvas/programmer-guide/custom-nodes/custom-nodeable-nodes.md:193]: REGISTER_SCRIPTCANVAS_AUTOGEN_FUNCTION is for custom function node which is covered by a different topic

REGISTER_SCRIPTCANVAS_AUTOGEN_GRAMMAR is for internal usage right now only, and some use case is related to deprecation node.

File: content/docs/user-guide/scripting/script-canvas/programmer-guide/custom-nodes/custom-nodeable-nodes.md [hunk 6]
```diff
@@ -168,88 +140,163 @@ There are three critical parts that every Script Canvas nodeable header file nee
...
+    {{< note >}}
+    Sanitized Gem target name should contain letters and numbers only. In **StartingPointInput** example, it is `StartingPointInputStatic` which refers to `StartingPointInput.Static` target.
+    {{< /note >}}
+   
+    For example, in `StartingPointInputGem.cpp`
+  
+    ```cpp
+    #include <AutoGenFunctionRegistry.generated.h>
+    
+    REGISTER_SCRIPTCANVAS_AUTOGEN_FUNCTION(MyGemStatic)
```
[the remark this answers is not in the document]
Comment (onecent1101) [reply content/docs/user-guide/scripting/script-canvas/programmer-guide/custom-nodes/custom-nodeable-nodes.md]: good catch, copy paste mistake, for nodeable usage, we just need Nodeable related header and macros

**Pull request merged**